doc/avx_kernels/implementation.rst

```diff
@@ -100,13 +100,13 @@ When such a function is given, we can choose a proper function to call for a giv
 As the wire the gate applies to for the internal function should be a compile-time parameter, we need a switch-case statement for calling internal functions.
```

Comment:
I think we may need to reword this a little.

Maybe something like:
```
As we require the wire index to be a compile-time parameter...
```

Reply:
Now changed to:
> Note that we used a switch-case statement for calling internal functions as the wire index must be a compile-time template parameter for `applyInternal`.

Reply:
Perfect.